### ISSUE TITLE
Installing kaltura-ffmpeg from RHEL repo for thumbnails support

### ISSUE THREAD

**Issue by KuroNeko-san** [kaltura/nginx-vod-module#409]
Hello @erankor and other developers of this great soft!

I trying to use thumbnail function of nginx-vod-module and trying to install kaltura-ffmpeg for it.

```
# yum --enablerepo=kaltura,kaltura-noarch install kaltura-ffmpeg kaltura-ffmpeg-devel

Loaded plugins: fastestmirror
Setting up Install Process
Loading mirror speeds from cached hostfile
 * base: mirror.awanti.com
 * extras: mirror.awanti.com
 * updates: mirror.awanti.com
Resolving Dependencies
--> Running transaction check
---> Package kaltura-ffmpeg.x86_64 0:2.7.2-2 will be installed
---> Package kaltura-ffmpeg-devel.x86_64 0:2.1.3-2 will be installed
--> Processing Dependency: kaltura-ffmpeg = 2.1.3 for package: kaltura-ffmpeg-devel-2.1.3-2.x86_64
--> Processing Dependency: openjpeg-devel for package: kaltura-ffmpeg-devel-2.1.3-2.x86_64
--> Processing Dependency: kaltura-x264-devel for package: kaltura-ffmpeg-devel-2.1.3-2.x86_64
--> Processing Dependency: libavfilter.so.3()(64bit) for package: kaltura-ffmpeg-devel-2.1.3-2.x86_64
--> Processing Dependency: libavdevice.so.55()(64bit) for package: kaltura-ffmpeg-devel-2.1.3-2.x86_64
--> Running transaction check
---> Package kaltura-ffmpeg-aux.x86_64 0:2.1.3-1 will be installed
---> Package kaltura-ffmpeg-devel.x86_64 0:2.1.3-2 will be installed
--> Processing Dependency: kaltura-ffmpeg = 2.1.3 for package: kaltura-ffmpeg-devel-2.1.3-2.x86_64
---> Package kaltura-x264-devel.x86_64 0:0.140-2.20140104 will be installed
---> Package openjpeg-devel.x86_64 0:1.3-11.el6 will be installed
--> Finished Dependency Resolution
Error: Package: kaltura-ffmpeg-devel-2.1.3-2.x86_64 (kaltura)
           Requires: kaltura-ffmpeg = 2.1.3
           Available: kaltura-ffmpeg-1.1.1-9.x86_64 (kaltura)
               kaltura-ffmpeg = 1.1.1-9
           Available: kaltura-ffmpeg-2.1.3-1.x86_64 (kaltura)
               kaltura-ffmpeg = 2.1.3-1
           Available: kaltura-ffmpeg-2.1.3-7.x86_64 (kaltura)
               kaltura-ffmpeg = 2.1.3-7
           Available: kaltura-ffmpeg-2.1.3-8.x86_64 (kaltura)
               kaltura-ffmpeg = 2.1.3-8
           Available: kaltura-ffmpeg-2.1.3-9.x86_64 (kaltura)
               kaltura-ffmpeg = 2.1.3-9
           Available: kaltura-ffmpeg-2.7.2-1.x86_64 (kaltura)
               kaltura-ffmpeg = 2.7.2-1
           Installing: kaltura-ffmpeg-2.7.2-2.x86_64 (kaltura)
               kaltura-ffmpeg = 2.7.2-2
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
```

Hm, dependency conflicts - version mismatch.

```
# yum --enablerepo=kaltura,kaltura-noarch install kaltura-ffmpeg-2.1.3-9.x86_64 kaltura-ffmpeg-devel

Loaded plugins: fastestmirror
Setting up Install Process
Loading mirror speeds from cached hostfile
 * base: mirror.awanti.com
 * extras: mirror.awanti.com
 * updates: mirror.awanti.com
Resolving Dependencies
--> Running transaction check
---> Package kaltura-ffmpeg.x86_64 0:2.1.3-9 will be installed
---> Package kaltura-ffmpeg-devel.x86_64 0:2.1.3-2 will be installed
--> Processing Dependency: openjpeg-devel for package: kaltura-ffmpeg-devel-2.1.3-2.x86_64
--> Processing Dependency: kaltura-x264-devel for package: kaltura-ffmpeg-devel-2.1.3-2.x86_64
--> Running transaction check
---> Package kaltura-x264-devel.x86_64 0:0.140-2.20140104 will be installed
---> Package openjpeg-devel.x86_64 0:1.3-11.el6 will be installed
--> Finished Dependency Resolution

Dependencies Resolved

======================================================================================================================================================================
 Package                                        Arch                             Version                                      Repository                         Size
======================================================================================================================================================================
Installing:
 kaltura-ffmpeg                                 x86_64                           2.1.3-9                                      kaltura                           4.9 M
 kaltura-ffmpeg-devel                           x86_64                           2.1.3-2                                      kaltura                           8.5 M
Installing for dependencies:
 kaltura-x264-devel                             x86_64                           0.140-2.20140104                             kaltura                            29 k
 openjpeg-devel                                 x86_64                           1.3-11.el6                                   base                               14 k

Transaction Summary
======================================================================================================================================================================
Install       4 Package(s)

Total size: 13 M
Total download size: 4.9 M
Installed size: 49 M
Is this ok [y/N]: y
Downloading Packages:
kaltura-ffmpeg-2.1.3-9.x86_64.rpm                                                                                                              | 4.9 MB     00:10
Running rpm_check_debug
Running Transaction Test


Transaction Check Error:
  file /opt/kaltura/ffmpeg-2.1.3/lib/pkgconfig/libavcodec.pc conflicts between attempted installs of kaltura-ffmpeg-2.1.3-9.x86_64 and kaltura-ffmpeg-devel-2.1.3-2.x86_64
  file /opt/kaltura/ffmpeg-2.1.3/lib/pkgconfig/libavdevice.pc conflicts between attempted installs of kaltura-ffmpeg-2.1.3-9.x86_64 and kaltura-ffmpeg-devel-2.1.3-2.x86_64
  file /opt/kaltura/ffmpeg-2.1.3/lib/pkgconfig/libavfilter.pc conflicts between attempted installs of kaltura-ffmpeg-2.1.3-9.x86_64 and kaltura-ffmpeg-devel-2.1.3-2.x86_64
  file /opt/kaltura/ffmpeg-2.1.3/lib/pkgconfig/libavformat.pc conflicts between attempted installs of kaltura-ffmpeg-2.1.3-9.x86_64 and kaltura-ffmpeg-devel-2.1.3-2.x86_64
  file /opt/kaltura/ffmpeg-2.1.3/share/man/man1/ffmpeg-all.1 conflicts between attempted installs of kaltura-ffmpeg-2.1.3-9.x86_64 and kaltura-ffmpeg-devel-2.1.3-2.x86_64
  file /opt/kaltura/ffmpeg-2.1.3/share/man/man1/ffmpeg-bitstream-filters.1 conflicts between attempted installs of kaltura-ffmpeg-2.1.3-9.x86_64 and kaltura-ffmpeg-devel-2.1.3-2.x86_64
  file /opt/kaltura/ffmpeg-2.1.3/share/man/man1/ffmpeg-codecs.1 conflicts between attempted installs of kaltura-ffmpeg-2.1.3-9.x86_64 and kaltura-ffmpeg-devel-2.1.3-2.x86_64
  file /opt/kaltura/ffmpeg-2.1.3/share/man/man1/ffmpeg-devices.1 conflicts between attempted installs of kaltura-ffmpeg-2.1.3-9.x86_64 and kaltura-ffmpeg-devel-2.1.3-2.x86_64
  file /opt/kaltura/ffmpeg-2.1.3/share/man/man1/ffmpeg-filters.1 conflicts between attempted installs of kaltura-ffmpeg-2.1.3-9.x86_64 and kaltura-ffmpeg-devel-2.1.3-2.x86_64
  file /opt/kaltura/ffmpeg-2.1.3/share/man/man1/ffmpeg-formats.1 conflicts between attempted installs of kaltura-ffmpeg-2.1.3-9.x86_64 and kaltura-ffmpeg-devel-2.1.3-2.x86_64
  file /opt/kaltura/ffmpeg-2.1.3/share/man/man1/ffmpeg-protocols.1 conflicts between attempted installs of kaltura-ffmpeg-2.1.3-9.x86_64 and kaltura-ffmpeg-devel-2.1.3-2.x86_64
  file /opt/kaltura/ffmpeg-2.1.3/share/man/man1/ffmpeg-resampler.1 conflicts between attempted installs of kaltura-ffmpeg-2.1.3-9.x86_64 and kaltura-ffmpeg-devel-2.1.3-2.x86_64
  file /opt/kaltura/ffmpeg-2.1.3/share/man/man1/ffmpeg-scaler.1 conflicts between attempted installs of kaltura-ffmpeg-2.1.3-9.x86_64 and kaltura-ffmpeg-devel-2.1.3-2.x86_64
  file /opt/kaltura/ffmpeg-2.1.3/share/man/man1/ffmpeg-utils.1 conflicts between attempted installs of kaltura-ffmpeg-2.1.3-9.x86_64 and kaltura-ffmpeg-devel-2.1.3-2.x86_64
  file /opt/kaltura/ffmpeg-2.1.3/share/man/man1/ffmpeg.1 conflicts between attempted installs of kaltura-ffmpeg-2.1.3-9.x86_64 and kaltura-ffmpeg-devel-2.1.3-2.x86_64
  file /opt/kaltura/ffmpeg-2.1.3/share/man/man1/ffplay-all.1 conflicts between attempted installs of kaltura-ffmpeg-2.1.3-9.x86_64 and kaltura-ffmpeg-devel-2.1.3-2.x86_64
  file /opt/kaltura/ffmpeg-2.1.3/share/man/man1/ffplay.1 conflicts between attempted installs of kaltura-ffmpeg-2.1.3-9.x86_64 and kaltura-ffmpeg-devel-2.1.3-2.x86_64
  file /opt/kaltura/ffmpeg-2.1.3/share/man/man1/ffprobe-all.1 conflicts between attempted installs of kaltura-ffmpeg-2.1.3-9.x86_64 and kaltura-ffmpeg-devel-2.1.3-2.x86_64
  file /opt/kaltura/ffmpeg-2.1.3/share/man/man1/ffprobe.1 conflicts between attempted installs of kaltura-ffmpeg-2.1.3-9.x86_64 and kaltura-ffmpeg-devel-2.1.3-2.x86_64
  file /opt/kaltura/ffmpeg-2.1.3/share/man/man1/ffserver-all.1 conflicts between attempted installs of kaltura-ffmpeg-2.1.3-9.x86_64 and kaltura-ffmpeg-devel-2.1.3-2.x86_64
  file /opt/kaltura/ffmpeg-2.1.3/share/man/man1/ffserver.1 conflicts between attempted installs of kaltura-ffmpeg-2.1.3-9.x86_64 and kaltura-ffmpeg-devel-2.1.3-2.x86_64

Error Summary
-------------

```

How to fix this ? Thanks.

**Comment by jessp01** [comment]
Hello,

The kaltura-ffmpeg-devel package is not needed for runtime, only for development [i.e, when compiling a new kaltura-ffmpeg package or when developing something that requires ffmpeg header and archive files].

Because it is not often used, it existed at an older version than the current kaltura-ffmpeg package [2.7.2-2] and so, when you tried to install both kaltura-ffmpeg and kaltura-ffmpeg-devel, it created a conflict. I just pushed kaltura-ffmpeg-devel-2.7.2-2.x86_64.rpm into the repo but since it is served from CDN, it might take some time for your edge to update. This shouldn't matter in your case however, because all you need is the kaltura-ffmpeg package.

To make a long story short, if using RHEL/CentOS 6, make sure /etc/yum.repos.d/kaltura.repo contains:
```
[Kaltura]
name = Kaltura Server
baseurl = http://installrepo.kaltura.org/releases/latest/RPMS/$basearch/
gpgkey = http://installrepo.kaltura.org/releases/RPM-GPG-KEY-kaltura
gpgcheck = 1
enabled = 1
```

or, if using RHEL/CentOS 7:
```
[Kaltura]
name = Kaltura Server
baseurl = http://installrepo.kaltura.org/releases/rhel7/RPMS/$basearch/
gpgkey = http://installrepo.kaltura.org/releases/RPM-GPG-KEY-kaltura
gpgcheck = 1
enabled = 1
```

And run:
\# yum install kaltura-ffmpeg

You should be getting version kaltura-ffmpeg-2.7.2-2.x86_64.

Thanks for reporting the issue!

**Comment by KuroNeko-san** [comment]
I tried to install only kaltura-ffmpeg, but ./configure seems to ignore "libavcodec" in this case.
I think kaltura-ffmpeg-devel needed for this. I'm right ?

For now "kaltura-ffmpeg-devel" show error about "kaltura-lame":

```
# yum --enablerepo=kaltura,kaltura-noarch install kaltura-ffmpeg-devel

Loaded plugins: fastestmirror
Setting up Install Process
Loading mirror speeds from cached hostfile
 * base: mirror.yandex.ru
 * extras: mirror.yandex.ru
 * updates: mirror.yandex.ru
Resolving Dependencies
--> Running transaction check
---> Package kaltura-ffmpeg-devel.x86_64 0:2.7.2-2 will be installed
--> Processing Dependency: openjpeg-devel for package: kaltura-ffmpeg-devel-2.7.2-2.x86_64
--> Processing Dependency: kaltura-x264-devel for package: kaltura-ffmpeg-devel-2.7.2-2.x86_64
--> Processing Dependency: kaltura-lame-devel for package: kaltura-ffmpeg-devel-2.7.2-2.x86_64
--> Processing Dependency: kaltura-a52dec-devel for package: kaltura-ffmpeg-devel-2.7.2-2.x86_64
--> Running transaction check
---> Package kaltura-ffmpeg-devel.x86_64 0:2.7.2-2 will be installed
--> Processing Dependency: kaltura-a52dec-devel for package: kaltura-ffmpeg-devel-2.7.2-2.x86_64
---> Package kaltura-lame-devel.x86_64 0:3.99.5-2 will be installed
--> Processing Dependency: kaltura-lame = 3.99.5-2 for package: kaltura-lame-devel-3.99.5-2.x86_64
---> Package kaltura-x264-devel.x86_64 0:0.140-2.20140104 will be installed
---> Package openjpeg-devel.x86_64 0:1.3-11.el6 will be installed
--> Finished Dependency Resolution
Error: Package: kaltura-ffmpeg-devel-2.7.2-2.x86_64 (kaltura)
           Requires: kaltura-a52dec-devel
Error: Package: kaltura-lame-devel-3.99.5-2.x86_64 (kaltura)
           Requires: kaltura-lame = 3.99.5-2
           Available: kaltura-lame-3.99.5-3.x86_64 (kaltura)
               kaltura-lame = 3.99.5-3
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
```

Installed packages:

```
# rpm -qa | grep kaltura

kaltura-ffmpeg-2.7.2-2.x86_64
kaltura-release-11.8.0-1.noarch
kaltura-a52dec-0.7.4-11.x86_64
kaltura-x264-0.140-2.20140104.x86_64

```

Kaltura ENV

```
# cat /etc/profile.d/kaltura_ffmpeg.sh
PATH=$PATH:/opt/kaltura/ffmpeg-2.7.2/bin
export PATHi

LD_LIBRARY_PATH=/opt/kaltura/ffmpeg-2.7.2/lib
LIBRARY_PATH=/opt/kaltura/ffmpeg-2.7.2/lib
C_INCLUDE_PATH=/opt/kaltura/ffmpeg-2.7.2/include
export LD_LIBRARY_PATH LIBRARY_PATH C_INCLUDE_PATH
```

Configure log

```
# ./configure --with-http_stub_status_module --with-http_ssl_module --with-http_xslt_module --with-http_secure_link_module --with-http_flv_module --with-http_mp4_module --add-module=/usr/src/nginx-rtmp-module --add-module=/usr/src/nginx-vod-module --with-file-aio
checking for OS
 + Linux 2.6.32-573.22.1.el6.x86_64 x86_64
checking for C compiler ... found
 + using GNU C compiler
 + gcc version: 4.4.7 20120313 (Red Hat 4.4.7-16) (GCC)
checking for gcc -pipe switch ... found
checking for gcc builtin atomic operations ... found
checking for C99 variadic macros ... found
checking for gcc variadic macros ... found
checking for unistd.h ... found
checking for inttypes.h ... found
checking for limits.h ... found
checking for sys/filio.h ... not found
checking for sys/param.h ... found
checking for sys/mount.h ... found
checking for sys/statvfs.h ... found
checking for crypt.h ... found
checking for Linux specific features
checking for epoll ... found
checking for EPOLLRDHUP ... found
checking for O_PATH ... not found
checking for sendfile() ... found
checking for sendfile64() ... found
checking for sys/prctl.h ... found
checking for prctl(PR_SET_DUMPABLE) ... found
checking for sched_setaffinity() ... found
checking for crypt_r() ... found
checking for sys/vfs.h ... found
checking for nobody group ... found
checking for poll() ... found
checking for /dev/poll ... not found
checking for kqueue ... not found
checking for crypt() ... not found
checking for crypt() in libcrypt ... found
checking for F_READAHEAD ... not found
checking for posix_fadvise() ... found
checking for O_DIRECT ... found
checking for F_NOCACHE ... not found
checking for directio() ... not found
checking for statfs() ... found
checking for statvfs() ... found
checking for dlopen() ... not found
checking for dlopen() in libdl ... found
checking for sched_yield() ... found
checking for SO_SETFIB ... not found
checking for SO_ACCEPTFILTER ... not found
checking for TCP_DEFER_ACCEPT ... found
checking for TCP_KEEPIDLE ... found
checking for TCP_FASTOPEN ... not found
checking for TCP_INFO ... found
checking for accept4() ... found
checking for kqueue AIO support ... not found
checking for Linux AIO support ... found
checking for int size ... 4 bytes
checking for long size ... 8 bytes
checking for long long size ... 8 bytes
checking for void * size ... 8 bytes
checking for uint64_t ... found
checking for sig_atomic_t ... found
checking for sig_atomic_t size ... 4 bytes
checking for socklen_t ... found
checking for in_addr_t ... found
checking for in_port_t ... found
checking for rlim_t ... found
checking for uintptr_t ... uintptr_t found
checking for system byte ordering ... little endian
checking for size_t size ... 8 bytes
checking for off_t size ... 8 bytes
checking for time_t size ... 8 bytes
checking for setproctitle() ... not found
checking for pread() ... found
checking for pwrite() ... found
checking for sys_nerr ... found
checking for localtime_r() ... found
checking for posix_memalign() ... found
checking for memalign() ... found
checking for mmap(MAP_ANON|MAP_SHARED) ... found
checking for mmap("/dev/zero", MAP_SHARED) ... found
checking for System V shared memory ... found
checking for POSIX semaphores ... not found
checking for POSIX semaphores in libpthread ... found
checking for struct msghdr.msg_control ... found
checking for ioctl(FIONBIO) ... found
checking for struct tm.tm_gmtoff ... found
checking for struct dirent.d_namlen ... not found
checking for struct dirent.d_type ... found
checking for sysconf(_SC_NPROCESSORS_ONLN) ... found
checking for openat(), fstatat() ... found
checking for getaddrinfo() ... found
configuring additional modules
adding module in /usr/src/nginx-rtmp-module
 + ngx_rtmp_module was configured
adding module in /usr/src/nginx-vod-module
checking for OpenSSL EVP library ... found
checking for clock_gettime() ... not found
checking for clock_gettime() in librt ... found
checking for libavcodec ... not found
checking for libavfilter ... not found
 + ngx_http_vod_module was configured
checking for PCRE library ... found
checking for PCRE JIT support ... not found
checking for OpenSSL library ... found
checking for zlib library ... found
checking for libxslt ... found
checking for libexslt ... found
creating objs/Makefile

Configuration summary
  + using system PCRE library
  + using system OpenSSL library
  + md5: using OpenSSL library
  + sha1: using OpenSSL library
  + using system zlib library

  nginx path prefix: "/usr/local/nginx"
  nginx binary file: "/usr/local/nginx/sbin/nginx"
  nginx configuration prefix: "/usr/local/nginx/conf"
  nginx configuration file: "/usr/local/nginx/conf/nginx.conf"
  nginx pid file: "/usr/local/nginx/logs/nginx.pid"
  nginx error log file: "/usr/local/nginx/logs/error.log"
  nginx http access log file: "/usr/local/nginx/logs/access.log"
  nginx http client request body temporary files: "client_body_temp"
  nginx http proxy temporary files: "proxy_temp"
  nginx http fastcgi temporary files: "fastcgi_temp"
  nginx http uwsgi temporary files: "uwsgi_temp"
  nginx http scgi temporary files: "scgi_temp"
```

If i make this and enable thumbnails, i'll get error:

```
location /thumbs-test/ {
    vod thumb;
    vod_mode local;

    alias /data/vod;
}

# service nginx restart
nginx: [emerg] invalid value "thumb" in "vod" directive, it must be one of: none,dash,hds,hls,mss in /usr/local/nginx/conf/conf.d/nginx-http.conf:121
nginx: configuration file /usr/local/nginx/conf/nginx.conf test failed
```

Where i'm wrong ?

**Comment by jessp01** [comment]
Hi @KuroNeko-san,

According to your configure output, 
checking for libavcodec ... not found

Like I said, if you wish to compile Nginx with the vod-module yourself, you DO need kaltura-ffmepg-devel.
Please install it like so for RHEL/CentOS 6:
\# rpm -ihv http://installrepo.kaltura.org/releases/12.0.0/RPMS/x86_64/kaltura-ffmpeg-devel-2.7.2-2.x86_64.rpm

or for RHEL/CentOS 7:
\# rpm -ihv http://installrepo.origin.kaltura.org/releases/rhel7/RPMS/x86_64/kaltura-ffmpeg-devel-2.7.2-2.x86_64.rpm

Installing these RPMs directly is only needed because I'm not sure your CDN edge is updated with the latest packages, by tomorrow, yum install kaltura-ffmpeg-devel should be enough.

Then run the configure script again, exporting the ENV vars as you've done before, and make sure you see:
```
checking for libavcodec ... found
```
If you do, great, continue to make && make install
If you do not, check objs/autoconf.err for errors.

When the build is done, also run:
\# ldd objs/nginx|grep ffmpeg
and make sure you see:
```
libavcodec.so.56 => /opt/kaltura/ffmpeg-2.7.2/lib/libavcodec.so.56
```
in the output, then it should work correctly.

**Comment by KuroNeko-san** [comment]
Yes, yes, yes, i know that i needed kaltura-ffmpeg-devel for compilation, but i can't install it because of kaltura-lame. But previously you say i need kaltura-ffmpeg, not kaltura-ffmpeg-devel.

> This shouldn't matter in your case however, because all you need is the kaltura-ffmpeg package.

This is mistake, i see.

```
# yum --enablerepo=kaltura,kaltura-noarch install kaltura-ffmpeg-devel

Loaded plugins: fastestmirror
Setting up Install Process
Loading mirror speeds from cached hostfile
 * base: mirror.yandex.ru
 * extras: mirror.yandex.ru
 * updates: mirror.yandex.ru
Resolving Dependencies
--> Running transaction check
---> Package kaltura-ffmpeg-devel.x86_64 0:2.7.2-2 will be installed
--> Processing Dependency: kaltura-ffmpeg = 2.7.2 for package: kaltura-ffmpeg-devel-2.7.2-2.x86_64
--> Processing Dependency: openjpeg-devel for package: kaltura-ffmpeg-devel-2.7.2-2.x86_64
--> Processing Dependency: kaltura-x264-devel for package: kaltura-ffmpeg-devel-2.7.2-2.x86_64
--> Processing Dependency: kaltura-x264 for package: kaltura-ffmpeg-devel-2.7.2-2.x86_64
--> Processing Dependency: kaltura-lame-devel for package: kaltura-ffmpeg-devel-2.7.2-2.x86_64
--> Processing Dependency: kaltura-a52dec-devel for package: kaltura-ffmpeg-devel-2.7.2-2.x86_64
--> Processing Dependency: libx264.so.140()(64bit) for package: kaltura-ffmpeg-devel-2.7.2-2.x86_64
--> Running transaction check
---> Package kaltura-ffmpeg.x86_64 0:2.7.2-2 will be installed
--> Processing Dependency: kaltura-a52dec for package: kaltura-ffmpeg-2.7.2-2.x86_64
---> Package kaltura-ffmpeg-devel.x86_64 0:2.7.2-2 will be installed
--> Processing Dependency: kaltura-a52dec-devel for package: kaltura-ffmpeg-devel-2.7.2-2.x86_64
---> Package kaltura-lame-devel.x86_64 0:3.99.5-2 will be installed
--> Processing Dependency: kaltura-lame = 3.99.5-2 for package: kaltura-lame-devel-3.99.5-2.x86_64
---> Package kaltura-x264.x86_64 0:0.140-2.20140104 will be installed
---> Package kaltura-x264-devel.x86_64 0:0.140-2.20140104 will be installed
---> Package openjpeg-devel.x86_64 0:1.3-11.el6 will be installed
--> Running transaction check
---> Package kaltura-a52dec.x86_64 0:0.7.4-11 will be installed
---> Package kaltura-ffmpeg-devel.x86_64 0:2.7.2-2 will be installed
--> Processing Dependency: kaltura-a52dec-devel for package: kaltura-ffmpeg-devel-2.7.2-2.x86_64
---> Package kaltura-lame-devel.x86_64 0:3.99.5-2 will be installed
--> Processing Dependency: kaltura-lame = 3.99.5-2 for package: kaltura-lame-devel-3.99.5-2.x86_64
--> Finished Dependency Resolution
Error: Package: kaltura-ffmpeg-devel-2.7.2-2.x86_64 (kaltura)
           Requires: kaltura-a52dec-devel
Error: Package: kaltura-lame-devel-3.99.5-2.x86_64 (kaltura)
           Requires: kaltura-lame = 3.99.5-2
           Available: kaltura-lame-3.99.5-3.x86_64 (kaltura)
               kaltura-lame = 3.99.5-3
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
```

kaltura-ffmpeg-devel-2.7.2-2.x86_64 requires kaltura-lame-3.99.5-**2**, available kaltura-lame-3.99.5-**3**

No changes if i use "rpm --ihv" instead of "yum install".

**Comment by jessp01** [comment]
Hi @KuroNeko-san,

What I meant is you do not need kaltura-ffmpeg-devel for runtime:
"The kaltura-ffmpeg-devel package is not needed for runtime, only for development [i.e, when compiling a new kaltura-ffmpeg package or when developing something that requires ffmpeg header and archive files]."

I did not understand from your original post that you intend to compile it yourself, sorry...

At any rate, I'll make sure all devel packages kaltura-ffmpeg-devel requires are available in a moment. To make sure I tested with the same distro version as you, which one are you using? 6 or 7?

Thanks,

**Comment by KuroNeko-san** [comment]
I'm using CentOS 6
```
# cat /etc/centos-release
CentOS release 6.7 (Final)
```

Thanks for you time and support.

**Comment by jessp01** [comment]
Hi @KuroNeko-san,

No need to apologise, I got you:)

Everything should be sorted out now. Please set the following in /etc/yum.repos.d/kaltura.list:
```
[Kaltura]
name = Kaltura Server
baseurl = http://installrepo.origin.kaltura.org/releases/nightly/RPMS/$basearch/
gpgkey = http://installrepo.kaltura.org/releases/RPM-GPG-KEY-kaltura
gpgcheck = 1
enabled = 1

[Kaltura-noarch]
name = Kaltura Server arch independent
baseurl = http://installrepo.origin.kaltura.org/releases/nightly/RPMS/noarch
gpgkey = http://installrepo.kaltura.org/releases/RPM-GPG-KEY-kaltura
gpgcheck = 1
enabled = 1
```

This will fetch the packages from the origin server and would avoid CDN issues.
Then, you should be able to do:
```
yum install kaltura-ffmpeg-devel
```
and continue with your build.
However, like I said before, you can also simply do:
```
yum update kaltura-nginx
```
And get a pre-build Nginx with this support already enabled.
Please let me know as to your progress.
Thanks,

**Comment by KuroNeko-san** [comment]
**It works for now, thanks!**

p.s. i can't simply install "kaltura-nginx" because i need nginx-rtmp-module.

**Comment by jessp01** [comment]
Thanks for the update, glad we're good:)
Understood about needing to recompile for the RTMP module. The newer versions of Nginx support DSO so hopefully, in the near future, you'll just be able to dynamically load additional modules into the precompiled Nginx from kaltura-nginx but for the current version, that is indeed not possible.
